pkg/resource/obtenantbackuppolicy_manager.go

```diff
@@ -0,0 +1,216 @@
+			}
+			m.BackupPolicy.Status.TenantInfo = tenant
+		}
+		err := m.syncLatestJobs()
+		if err != nil {
+			return err
+		}
+		backupPath := m.getBackupDestPath()
+		latestFull, err := m.getLatestBackupJobOfTypeAndPath(constants.BackupJobTypeFull, backupPath)
+		if err != nil {
+			return err
+		}
+		latestIncr, err := m.getLatestBackupJobOfTypeAndPath(constants.BackupJobTypeIncr, backupPath)
+		if err != nil {
+			return err
+		}
+		m.BackupPolicy.Status.LatestFullBackupJob = latestFull
+		m.BackupPolicy.Status.LatestIncrementalJob = latestIncr
+
+		if latestFull == nil || latestFull.Status == "CANCELED" {
+			m.BackupPolicy.Status.NextFull = time.Now().Format(time.DateTime)
+		} else if latestFull.Status == "COMPLETED" {
+			fullCron, err := cron.ParseStandard(m.BackupPolicy.Spec.DataBackup.FullCrontab)
+			if err != nil {
+				return err
+			}
+			var lastFullBackupFinishedAt time.Time
+			if latestFull.EndTimestamp != nil {
+				lastFullBackupFinishedAt, err = time.ParseInLocation(time.DateTime, *latestFull.EndTimestamp, time.Local)
+			}
+			nextFull := fullCron.Next(lastFullBackupFinishedAt)
+			m.BackupPolicy.Status.NextFull = nextFull.Format(time.DateTime)
+			if nextFull.After(time.Now()) {
+				incrCron, err := cron.ParseStandard(m.BackupPolicy.Spec.DataBackup.IncrementalCrontab)
+				if err != nil {
+					return err
+				}
+				if latestIncr == nil || latestIncr.Status == "CANCELED" {
+					m.BackupPolicy.Status.NextIncremental = time.Now().Format(time.DateTime)
+				} else if latestIncr.Status == "COMPLETED" {
+					var lastIncrBackupFinishedAt time.Time
+					if latestIncr.EndTimestamp != nil {
+						lastIncrBackupFinishedAt, err = time.ParseInLocation(time.DateTime, *latestIncr.EndTimestamp, time.Local)
+					}
+					m.BackupPolicy.Status.NextIncremental = incrCron.Next(lastIncrBackupFinishedAt).Format(time.DateTime)
+				}
+			}
+		}
+	}
+
+	return m.retryUpdateStatus()
+}
+
+func (m *ObTenantBackupPolicyManager) GetTaskFunc(name string) (func() error, error) {
+	switch name {
+	case taskname.ConfigureServerForBackup:
+		return m.ConfigureServerForBackup, nil
+	case taskname.GetTenantInfo:
+		return m.GetTenantInfo, nil
+	case taskname.StartBackupJob:
+		return m.StartBackup, nil
+	case taskname.StopBackupJob:
+		return m.StopBackup, nil
+	case taskname.CheckAndSpawnJobs:
+		return m.CheckAndSpawnJobs, nil
+	case taskname.CleanOldBackupJobs:
+		return m.CleanOldBackupJobs, nil
+	case taskname.PauseBackup:
+		return m.PauseBackup, nil
+	case taskname.ResumeBackup:
+		return m.ResumeBackup, nil
+	default:
+		return nil, errors.Errorf("unknown task name %s", name)
+	}
+}
+
+func (m *ObTenantBackupPolicyManager) GetTaskFlow() (*task.TaskFlow, error) {
+	// exists unfinished task flow, return the last task flow
+	if m.BackupPolicy.Status.OperationContext != nil {
+		return task.NewTaskFlow(m.BackupPolicy.Status.OperationContext), nil
+	}
+	status := m.BackupPolicy.Status.Status
+	// get task flow depending on BackupPolicy status
```

Comment:
what's the behavior when delete a policy, maybe still need a task flow to cancel all the jobs

Reply:
In ObTenantBackupPolicyManager.`CheckAndUpdateFinalizers`, all backup, log archive and clean jobs will be canceled. Deletion logic is finalizers' responsibility.